home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/CooperationApiImpl.java

```diff
@@ -67,7 +67,7 @@ public Response createHouse(Long cooperationId, @Valid CreateHouse createHouse)
     @PreAuthorize(GET_COOPERATION_PERMISSION)
     @Override
     public Response getCooperation(Long cooperationId) {
-        CooperationDto readCoopDto = cooperationService.getCooperationById(cooperationId);
+        CooperationDto readCoopDto = (CooperationDto) queryApiService.getOne(uriInfo, cooperationService);
```

Comment:
Can we move this cast into getOne method?

Reply:
It is impossible for now because of Java behavior which ignores the generic of the method in the case when the generic of the class is not defined. A separate task will be created to refactor QueryApiService and API implementations to avoid of using entity services inside QueryApiService.